src/iceberg/file_writer.h

```diff
@@ -65,6 +67,20 @@ class ICEBERG_EXPORT Writer {
   ///
   /// \return Status of write results.
   virtual Status Write(ArrowArray data) = 0;
+
+  /// \brief Get the file statistics.
+  /// Only valid after the file is closed.
+  virtual Metrics metrics() = 0;
```

Comment:
I'm not sure if it is better to return std::optional<Metrics> and std::optional<int64_t> for metrics and length. Let me know what you think.

Reply:
Actually it should be , internal implement will check whether the stream is closed, if not return null